hnn_core/extracellular.py

```diff
@@ -65,23 +65,30 @@ def _transfer_resistance(section, electrode_pos, sigma, method):
     Returns
     -------
     vres : list
-        The resistance at each section.
+        The transfer resistance at each segment of the section.
     """
     electrode_pos = np.array(electrode_pos)  # electrode position to Numpy
+    seg_gap = section.L / (section.nseg + 1)
+    seg_ctr = np.zeros((section.nseg, 3))
 
-    start = np.array([section.x3d(0), section.y3d(0), section.z3d(0)])
-    end = np.array([section.x3d(1), section.y3d(1), section.z3d(1)])
+    sec_start = np.array([section.x3d(0), section.y3d(0), section.z3d(0)])
+    sec_end = np.array([section.x3d(1), section.y3d(1), section.z3d(1)])
+    sec_ori = sec_end - sec_start
+
+    for ii, seg in enumerate(section):
+        seg_ctr[ii, :] = sec_start + seg.x * sec_ori
 
     if method == 'psa':
 
-        mid = (start + end) / 2.
+        # mid = (start + end) / 2.
 
-        # distance from section midpoint to electrode
-        dis = norm(electrode_pos - mid)
+        # distance from segment midpoint to electrode
+        dis = norm(np.tile(electrode_pos, (section.nseg, 1)) - seg_ctr,
+                   axis=1)
 
-        # setting radius limit
-        if dis < section.diam / 2.0:
-            dis = section.diam / 2.0 + 0.1
+        # # setting radius limit
+        # if dis < section.diam / 2.0:
+        #     dis = section.diam / 2.0 + 0.1
```

Comment:
This needs to be discussed. It's related to the discussion/Issue related to cell morphology and spacing.